src/include/type/expression/expression_defs.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <cstdint>
+
+namespace terrier::type::expression {
+
+/**
+ * All possible expression types.
+ * TODO(WAN): Do we really need some of this stuff? e.g. HASH_RANGE for elastic
+ */
+enum class ExpressionType : uint8_t {
+  INVALID,
```

Comment:
Not clear which ones we need to support and which not. What some of the types mean (semantics) is also not clear. Can we flag the questionable ones for discussion / review?